tests/api/v1/collectorGroups/put.js

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * tests/api/v1/collectorGroups/put.js
+ */
+'use strict'; // eslint-disable-line strict
+const supertest = require('supertest');
+const api = supertest(require('../../../../index').app);
+const httpStatus = require('../../../../api/v1/constants').httpStatus;
+const tu = require('../../../testUtils');
+const u = require('./utils');
+const Collector = tu.db.Collector;
+const CollectorGroup = tu.db.CollectorGroup;
+const expect = require('chai').expect;
+
+describe('tests/api/v1/collectorGroups/put.js >', () => {
+  let user;
+  let token;
+  let token2;
+  let collector1;
+  let collector2;
+  let collector3;
+  let cg = { name: '' };
+
+  before((done) => {
+    tu.createUserAndToken()
+      .then((ut) => {
+        user = ut.user;
+        token = ut.token;
+        return tu.createSecondUser();
+      })
+      .then((user2) => tu.createTokenFromUserName(user2.get('name')))
+      .then((t) => (token2 = t))
+      .then(() => done())
+      .catch(done);
+  });
+
+  beforeEach(() => {
+    collector1 = u.getCollectorToCreate();
+    collector1.name += '1';
+    Collector.create(collector1);
+    collector2 = u.getCollectorToCreate();
+    collector2.name += '2';
+    Collector.create(collector2);
+    collector3 = u.getCollectorToCreate();
+    collector3.name += '3';
+    Collector.create(collector3);
+  });
+
+  beforeEach((done) => {
+    CollectorGroup.createCollectorGroup({
+      name: `${tu.namePrefix}cg`,
+      description: 'desc',
+      createdBy: user.id,
+    })
+      .then((created) => {
+        cg = created;
+        done();
+      })
+      .catch(done);
+  });
+
+  afterEach(u.forceDelete);
+
+  after(tu.forceDeleteUser);
+
+  it('put with empty collector list', (done) => {
+    api.put(`/v1/collectorGroups/${cg.name}`)
+      .set('Authorization', token)
+      .send({ name: cg.name, description: 'new desc', collectors: [] })
+      .expect(httpStatus.OK)
+      .end((err, res) => {
+        if (err) {
+          return done(err);
+        }
+
+        expect(res.body).to.have.property('description', 'new desc');
+        expect(res.body.collectors).to.be.empty;
+        return done();
+      });
+  });
```

Comment:
Should also test the case where there are already collectors set.

Reply:
oh yeah - i think put and patch should both be *replacing* whatever is there, so should not fail if it is sending a collector which is already assigned to it (but *should* fail if a collector it is assigning is already assigned to a *different* collector group)